In [12]:
import math
import random
import statistics

zadanie 2
a)

In [3]:
vector1 = [3, 8, 9, 10, 12]
vector2 = [8, 7, 7, 5, 6]

sumowanie po współrzędnych:

In [4]:
vectors_sum = []

for i in range(len(vector1)):
    vectors_sum.append(vector1[i] + vector2[i])
print(vectors_sum)

[11, 15, 16, 15, 18]


iloczyn po współrzędnych:

In [5]:
vectors_product = []

for i in range(len(vector1)):
    vectors_product.append(vector1[i] * vector2[i])
print(vectors_product)

[24, 56, 63, 50, 72]


b) iloczyn skalarny

In [6]:
vectors_scalar_product = 0

for i in range(len(vector1)):
    vectors_scalar_product += vector1[i] * vector2[i]
print(vectors_scalar_product)

265


c) Dla powyższych wektorów podaj ich długości euklidesowe (długość wektora jako strzałki w
przestrzeni)

In [8]:
def make_euclidean_length(vector):
    result = 0
    for number in vector:
        result += number ** 2
    return math.sqrt(result)

print(make_euclidean_length(vector1))
print(make_euclidean_length(vector2))

19.949937343260004
14.933184523068078


d) Stwórz wektor 50 losowych liczb z zakresu od 1 do 100.

In [9]:
random_numbers = [random.randint(1, 100) for i in range(0, 50)]
print(random_numbers)

[75, 11, 28, 13, 36, 55, 93, 48, 89, 97, 47, 94, 9, 19, 93, 26, 92, 52, 68, 48, 60, 25, 15, 86, 80, 28, 40, 38, 5, 9, 17, 18, 65, 83, 43, 3, 12, 30, 74, 48, 71, 17, 65, 64, 95, 100, 3, 80, 94, 52]


e) Dla wektora z punktu (d) policz średnią z wszystkich jego liczb, min, max oraz odchylenie
standardowe.

In [24]:
random_numbers_min = min(random_numbers)
random_numbers_max = max(random_numbers)
random_numbers_mean = statistics.mean(random_numbers)

print("min: " + str(random_numbers_min))
print("max: " + str(random_numbers_max))
print("mean: " + str(random_numbers_mean))

min: 3
max: 100
mean: 50.26


f) Dokonaj normalizacji wektora z podpunktu (d)

In [25]:
random_numbers_normalized = []
for number in random_numbers:
    normalized_number = (number - random_numbers_min)/(random_numbers_max - random_numbers_min)
    random_numbers_normalized.append(normalized_number)

print(random_numbers_normalized)
print(random_numbers_normalized[45])

[0.7422680412371134, 0.08247422680412371, 0.25773195876288657, 0.10309278350515463, 0.3402061855670103, 0.5360824742268041, 0.9278350515463918, 0.4639175257731959, 0.8865979381443299, 0.9690721649484536, 0.4536082474226804, 0.9381443298969072, 0.061855670103092786, 0.16494845360824742, 0.9278350515463918, 0.23711340206185566, 0.9175257731958762, 0.5051546391752577, 0.6701030927835051, 0.4639175257731959, 0.5876288659793815, 0.2268041237113402, 0.12371134020618557, 0.8556701030927835, 0.7938144329896907, 0.25773195876288657, 0.38144329896907214, 0.36082474226804123, 0.020618556701030927, 0.061855670103092786, 0.14432989690721648, 0.15463917525773196, 0.6391752577319587, 0.8247422680412371, 0.41237113402061853, 0.0, 0.09278350515463918, 0.27835051546391754, 0.7319587628865979, 0.4639175257731959, 0.7010309278350515, 0.14432989690721648, 0.6391752577319587, 0.6288659793814433, 0.9484536082474226, 1.0, 0.0, 0.7938144329896907, 0.9381443298969072, 0.5051546391752577]
1.0


W oryginalnym wektorze jakie było max?
    max było równe 100
Na której pozycji stało?
    stało na pozycji 45
Jaka liczba stoi na tej pozycji w nowym wektorze?
    teraz stoi 1.0

g) Dokonaj standaryzacji wektora z podpunktu (d).

In [26]:
random_numbers_standardized = []
random_numbers_standard_deviation = statistics.stdev(random_numbers)

for number in random_numbers:
    standardized_number = (number - random_numbers_mean) / random_numbers_standard_deviation
    random_numbers_standardized.append(standardized_number)

print(random_numbers_standardized)

[0.7977882377307417, -1.2660131856632544, -0.7178159325742243, -1.201519391182192, -0.4598407546499747, 0.1528502929201179, 1.3782323880603031, -0.07287798776360042, 1.2492447990981783, 1.507219977022428, -0.10512488500413161, 1.4104792853008343, -1.3305069801443168, -1.008038007739005, 1.3782323880603031, -0.7823097270552866, 1.345985490819772, 0.05610960119852434, 0.5720599570470234, -0.07287798776360042, 0.3140847791227739, -0.8145566242958178, -1.1370255967011298, 1.1525041073765847, 0.9590227239333976, -0.7178159325742243, -0.33085316568784995, -0.39534696016891235, -1.4594945691064416, -1.3305069801443168, -1.0725318022200674, -1.0402849049795362, 0.4753192653254298, 1.0557634156549913, -0.23411247396625637, -1.523988363587504, -1.2337662884227232, -0.6533221380931619, 0.7655413404902105, -0.07287798776360042, 0.668800648768617, -1.0725318022200674, 0.4753192653254298, 0.44307236808489864, 1.4427261825413655, 1.6039606687440215, -1.523988363587504, 0.9590227239333976, 1.410479285

In [36]:
print(statistics.mean(random_numbers_standardized))
print(statistics.stdev(random_numbers_standardized))

6.439293542825907e-17
0.9999999999999999


Jaką średnią i odchylenie standardowe ma nowy wektor z?
    sredni to 6.439293542825907e-17
    odchylenie standardowe to 0.9999999999999999

h) Dokonaj dyskretyzacji wektora z podpunktu (d). Wektor liczb zamieni się na wektora
napisów określających przedziały: [0, 10), [10, 20), …, [90,100]. Inaczej mówiąc: liczba
zamieni się na przedział – taki, w jakim się ta liczba znajduje. Wyświetl ten wektor.

In [38]:
random_numbers_discretized = []

def number_discretization(numb):
    bottom = numb // 10 * 10 if numb < 100 else 90
    print(numb)
    print(bottom)
    print("-------------")

for number in random_numbers:
    number_discretization(number)

75
70
-------------
11
10
-------------
28
20
-------------
13
10
-------------
36
30
-------------
55
50
-------------
93
90
-------------
48
40
-------------
89
80
-------------
97
90
-------------
47
40
-------------
94
90
-------------
9
0
-------------
19
10
-------------
93
90
-------------
26
20
-------------
92
90
-------------
52
50
-------------
68
60
-------------
48
40
-------------
60
60
-------------
25
20
-------------
15
10
-------------
86
80
-------------
80
80
-------------
28
20
-------------
40
40
-------------
38
30
-------------
5
0
-------------
9
0
-------------
17
10
-------------
18
10
-------------
65
60
-------------
83
80
-------------
43
40
-------------
3
0
-------------
12
10
-------------
30
30
-------------
74
70
-------------
48
40
-------------
71
70
-------------
17
10
-------------
65
60
-------------
64
60
-------------
95
90
-------------
100
90
-------------
3
0
-------------
80
80
-------------
94
90
-------------
52
50
-------------
